src/http2/aws-http2-hpack.adb

```diff
@@ -308,54 +326,32 @@ package body AWS.HTTP2.HPACK is
                Settings.Set_Dynamic_Header_Table_Size (Natural (Idx));
             end if;
 
-         elsif BG4.B40 = B_II_No_Indexing then
-            --  Literal Header Field Never Indexed (RFC-7541 / 6.2.3)
-            --
-            --     0   1   2   3   4   5   6   7
-            --   +---+---+---+---+---+---+---+---+
-            --   | 0 | 0 | 0 | 1 |  Index (4+)   |
-            --   +---+---+-----------------------+
-            --
-            --  or
+         else
+            pragma Assert (BG4.B40 in B_II_No_Indexing | B_No_Indexing);
+            --  or just pragma Assert (BG4.B30 = 0);
+            --  The same decode processing for both RFC cases
+            --  (RFC-7541 / 6.2.3.).
             --
-            --     0   1   2   3   4   5   6   7
-            --   +---+---+---+---+---+---+---+---+
-            --   | 0 | 0 | 0 | 1 |     0         |
-            --   +---+---+-----------------------+
-
-            Idx := Get_Integer (4);
+            --  Literal Header Field Never Indexed (RFC-7541 / 6.2.3.).
+            --  No indexing - Indexed Name (RFC-7541 / 6.2.2.).
 
-            --  Get the name/value
-
-            declare
-               Name  : constant String :=
-                         (if Idx = 0
-                          then Get_String_Literal
-                          else Get_Indexed_Name (Idx));
-               Value : constant String := Get_String_Literal;
-            begin
-               AWS.Headers.Add (Headers, Name, Value);
-            end;
-
-            Data := True;
-
-         elsif BG4.B40 = B_No_Indexing then
-            --  No indexing - Indexed Name (RFC-7541 / 6.2.2)
             --
             --     0   1   2   3   4   5   6   7
             --   +---+---+---+---+---+---+---+---+
-            --   | 0 | 0 | 0 | 0 |  Index (4+)   |
+            --   | 0 | 0 | 0 | X |  Index (4+)   |
```

Comment:
I don't understand the X here. We really want the bit to be 0 ? Or maybe the diff is misleading... Can you comment?

Reply:
It is because when 3rd bit either 0 or 1, the processing is the same, see comment nearby. The difference is in the intermediate HTTP processors, proxy for example.

Reply:
I see, that's the case we've discussed some time ago.